### PR TITLE
Generated an abstract class for hosting the common method executed by…

### PR DESCRIPTION
**Pull request by nauar**
… the 2 YesNoApi task classes.

### CODE REVIEW

**Comment by dmarcas** [comment]
This branch needs to be updated with the last changes in master.

**Comment by dmarcas** [comment]
Hey Arnau, I've just seen the PR, but I think that this is not what we want to reach. The thing is to be able to get only one locustTask and make this dinamic according to the execution or to find another aproach that makes the code more reusable. This is a little bit tricky due that locust needs a locust task for every performance execution, but we want to rethink the way in how is working as far as possible.

File: src/main/java/locustTask/AbstractYesNoApiTask.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import io.restassured.RestAssured;
+import io.restassured.http.Method;
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import org.junit.Assert;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractYesNoApiTask extends AbstractTask {
```
Comment (dmarcas) [inline src/main/java/locustTask/AbstractYesNoApiTask.java]: With this class we avoid to rewrite the code of the execution but, what happens if we want to do another different reuqest with different baseURI? We will have to create another Abstract class to extends from locust4j and create the locustTask ... Now we will have to create one more classs ...

File: src/main/java/locustTask/AbstractYesNoApiTask.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    private static final Logger logger = LoggerFactory.getLogger(YesNoApi.class);
+
+    protected static final String baseURI = "https://yesno.wtf/api";
+
+    protected void commonTask(String URLGetParam, String expected) {
+        Response response = null;
```
Comment (dmarcas) [inline src/main/java/locustTask/AbstractYesNoApiTask.java]: Info: this is a demo code, in an ideal framework we should have a ServiceObject with the methods to be called.

Comment (nauar) [reply src/main/java/locustTask/AbstractYesNoApiTask.java]: Hey! Understood.

Let me plan something and I'll come back again with a new proposal.

Thanks for the explanation.

Comment (nauar) [reply src/main/java/locustTask/AbstractYesNoApiTask.java]: I have pushed a new proposal. The idea of SubTasks. You get the definition of all subtasks to be executed from any source (in this case a JSON file) and then one only task loops and executes every subtask.

**Review by therussiankid92 — COMMENTED** [review]
Thanks Arnau,
This is improving a lot our current codebase, making it cleaner.
To understand: for each new scenario we'll have to add a new locusttask, right?

**Comment by dmarcas** [comment]
Hey @therussiankid92 , I was talking with arnau just two hours ago ... in fact this changes doesn't work due that doesn't respect the locustTask as is defined in locust4j, and it will execute always all the tasks in every scenario iteration. By the other hand, the aproach is forcing to use a json to set some test values and maybe this is not ideal depending in the way that you want to work.

**Comment by therussiankid92** [comment]
Hi @nauar, 
We've updated master to reflect better what we're looking for from this request.
You can see now two feature files:
- One for the Functional Test
- One for the Performance

The Functional Test calls a method in the serviceobject/ForcedAnswer class.
The Performance Test calls LocustOperations which looks for the specified "locustTask" in the feature file. These are classes in the locusttask package (ForcedYes/ForcedNo), that are the same, but then they call the same serviceObject to be able to do the performance test.

We'd ideally want to avoid having to duplicate so much code (locustTasks, etc), just to do an extra scenario. Is there any way that you see we could achieve this? Ideally the main entry point to this should be the serviceObjects, and maybe have a generic locustTask. How do you see it?

Cheers,
Andrei

**Comment by therussiankid92** [comment]
@nauar -> any hint? We might use ourselves this time to improve this part!

**Pull request closed without merging**